lisa/transformers/scripts_execution_transformer.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Optional, Type
+
+from dataclasses_json import dataclass_json
+
+from lisa import schema
+from lisa.operating_system import Posix
+from lisa.transformers.deployment_transformer import (
+    DeploymentTransformer,
+    DeploymentTransformerSchema,
+)
+
+
+@dataclass_json()
+@dataclass
+class ScriptEntry:
+    script: str = ""
+    interpreter: str = ""
+    args: Optional[List[str]] = None
+
+
+@dataclass_json()
+@dataclass
+class ScriptsExecutionTransformerSchema(DeploymentTransformerSchema):
+    dependent_packages: List[str] = field(default_factory=list)
+    scripts: List[ScriptEntry] = field(default_factory=list)
+
+
+class ScriptExecutionResult:
+    output: str = ""
+    exit_code: Optional[int] = -1
+
+
+class ScriptsExecutionTransformer(DeploymentTransformer):
+    """
+    This Transformer is to execute scripts.
+
+    Sample runbook section:
+    - type: scripts_execution
+        phase: expanded
+        connection:
+        address: $(build_vm_address)
+        private_key_file: $(admin_private_key_file)
+        dependent_packages:
+        - git
+        scripts:
+        - script: "/tmp/waagent.sh"
+          interpreter: bash
+          args:
+            - "--flag"
+    """
+
+    __results_name = "scripts_execution_results"
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "scripts_execution"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return ScriptsExecutionTransformerSchema
+
+    @property
+    def _output_names(self) -> List[str]:
+        return [self.__results_name]
+
+    def _internal_run(self) -> Dict[str, Any]:
+        runbook: ScriptsExecutionTransformerSchema = self.runbook
+        os = self._node.os
+        if isinstance(os, Posix):
```

Comment:
Does it need to filter by Posix? The dependencies should be able to use by all oses.

Reply:
updated